autorest/azure/environments.go

```
@@ -0,0 +1,99 @@
+		ServiceManagementEndpoint: "https://management.core.usgovcloudapi.net/",
+		ResourceManagerEndpoint:   "https://management.usgovcloudapi.net",
+		ActiveDirectoryEndpoint:   "https://login.microsoftonline.com/",
+		GalleryEndpoint:           "https://gallery.usgovcloudapi.net/",
+		KeyVaultEndpoint:          "https://vault.azure.net/",
+		GraphEndpoint:             "https://graph.usgovcloudapi.net/",
+		StorageEndpointSuffix:     "core.usgovcloudapi.net",
+		SQLDatabaseDNSSuffix:      "database.usgovcloudapi.net",
+		TrafficManagerDNSSuffix:   "trafficmanager.net",
+		KeyVaultDNSSuffix:         "vault.azure.net",
+		ServiceBusEndpointSuffix:  "servicebus.usgovcloudapi.net",
+	}
+
+	// ChinaCloud is the cloud environment operated in China
+	ChinaCloud = Environment{
+		Name:                      "AzureChinaCloud",
+		ManagementPortalURL:       "https://manage.chinacloudapi.com/",
+		PublishSettingsURL:        "https://manage.chinacloudapi.com/publishsettings/index",
+		ServiceManagementEndpoint: "https://management.core.chinacloudapi.cn/",
+		ResourceManagerEndpoint:   "https://management.chinacloudapi.cn/",
+		ActiveDirectoryEndpoint:   "https://login.chinacloudapi.cn/?api-version=1.0",
+		GalleryEndpoint:           "https://gallery.chinacloudapi.cn/",
+		KeyVaultEndpoint:          "https://vault.azure.net/",
+		GraphEndpoint:             "https://graph.chinacloudapi.cn/",
+		StorageEndpointSuffix:     "core.chinacloudapi.cn",
+		SQLDatabaseDNSSuffix:      "database.chinacloudapi.cn",
+		TrafficManagerDNSSuffix:   "trafficmanager.cn",
+		KeyVaultDNSSuffix:         "vault.azure.net",
+		ServiceBusEndpointSuffix:  "servicebus.chinacloudapi.net",
+	}
+)
+
+// OAuthConfigForTenant returns an OAuthConfig with tenant specific urls
+func (env Environment) OAuthConfigForTenant(tenantID string) (*OAuthConfig, error) {
+	template := "%s/oauth2/%s?api-version=%s"
+	u, _ := url.Parse(env.ActiveDirectoryEndpoint)
```

I know it makes the code ugly but we should not ignore the errors.